weave/flow/eval.py

```diff
@@ -164,7 +178,23 @@ async def predict_and_score(
                 )
         try:
             model_start_time = time.time()
-            model_output = await async_call(model_predict, **model_predict_args)
+            model_call = None
+            if is_op(model_predict):
+                # I would expect this path to always be hit, but keeping the other
+                # path for backwards compatibility / safety
+                model_predict = as_op(model_predict)
+                if model_self is not None:
+                    model_predict_args = {
+                        **model_predict_args,
+                        "self": model_self,
+                    }
```

Comment:
nit: do you need to try all of this, or just the part below?

Reply:
just `async_call_op` or `async_call`

Reply:
I would make the try block just over that then